diff --git a/xrdpdev/xorg.conf b/xrdpdev/xorg.conf
index 24a7b05..24096a5 100644
--- a/xrdpdev/xorg.conf
+++ b/xrdpdev/xorg.conf
@@ -12,6 +12,7 @@ Section "ServerFlags"
     Option "DefaultServerLayout" "X11 Server"
     Option "DontVTSwitch" "on"
     Option "AutoAddDevices" "off"
+    Option "AutoAddGPU" "off"
 EndSection
 
 Section "Module"
@@ -59,6 +60,7 @@ Section "Screen"
     Device "Video Card (xrdpdev)"
     Monitor "Monitor"
     DefaultDepth 24
+    GPUDevice ""
     SubSection "Display"
         Depth 24
         Modes "640x480" "800x600" "1024x768" "1280x720" "1280x1024" "1600x900" "1920x1080"
